java/ql/src/semmle/code/java/frameworks/JaxWS.qll

```diff
@@ -955,3 +967,17 @@ private class VulnerableEntity extends XssSinkBarrier {
       ).getArgument(0)
   }
 }
+
+/**
+ * Model sources stemming from `ContainerRequestContext`.
+ */
+private class ContainerRequestContextModel extends SourceModelCsv {
+  override predicate row(string s) {
+    s =
+      ["javax", "jakarta"] + ".ws.rs.container;ContainerRequestContext;true;" +
+        [
+          "getAcceptableLanguages", "getAcceptableMediaTypes", "getCookies", "getEntityStream",
```

Comment:
Any reason to consider `getAcceptableLanguages` and `getAcceptableMediaTypes` as sources of taint, but not `getLanguage` and `getMediaType`? AFAIK all those come from headers, but I'm not sure about whether taint can be propagated through the types `Locale` and `MediaType` (my intuition says no, but I could be wrong, maybe it's worth to test it).

Reply:
Ok, I tested it and it seems that no exception gets thrown when arbitrary data gets passed to `Locale` and `MediaType`, and it can be later accessed with getters:

```java
package com.example;

import java.util.Locale;
import jakarta.ws.rs.core.MediaType;


public class App {
    public static void main(String[] args) {
        Locale l = new Locale("../../../'\" some injection;#--");
        System.out.println(l.toString());
        System.out.println(l.getDisplayCountry());
        System.out.println(l.getDisplayName());
        System.out.println(l.getDisplayLanguage());
        MediaType mt =
                new MediaType("../../..'\" some injection;#--", "../../../\"' some injection;#--");
        System.out.println(mt.toString());
        System.out.println(mt.getType());
        System.out.println(mt.getSubtype());
    }
}
```

So, it seems `getLanguage` and `getMediaType` should be added as sources too.

Reply:
Done